### PR TITLE
linkify only on post details page

### LINKED ISSUE #2153: Linkify post title and content can be conditional, e.g. only linkify when showing post details
right now in staging, `linkify` text happens in Comments, Messages and Posts (Helpboard and Post detailed page), it's better to apply `linkify` for Posts cases only in detailed Post page (Comments and Messages are ok, no need to change), this way will not slow down Helpboard loading, also when user click on Posts with links, `go to detailed Post` and `go to linked url` won't conflict with each other.

### PR DESCRIPTION
**Pull request by aditilamkhede**
### What does this pull request aim to achieve? 💯
convert text to url only when on post details page. If we are on helpborad(i.e. feed/posts) page, it should only be displayed as plain text.

_Please be concise and link any related issue(s):_ Resolves issue #2153 

- [x] I have checked that no one else is working on similar changes.
- [ ] I have read the latest [**README**](README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`**, or **`hotfix/<branch_name>`** if this is a hotfix, where `<branch_name>` briefly describes the issue(s) resolved and is prefixed with the issue number(s) (e.g. `1127-1130-update-git-branching-model`). The branch is created in a cloned version of this repo, **not a forked repo**.
- [x] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".
- [x] The title describes the issue(s) being addressed using format: **`<issue_numbers> - <brief_issue_description>`** (e.g. `#1127/#1130 - Update Git Branching Model`)

### CODE REVIEW

File: client/src/components/Feed/Post.js [hunk 1]
```diff
@@ -842,18 +843,26 @@ const Post = ({
...
   let finalContent = content;
   if (finalContent.length > CONTENT_LENGTH && !showComplete) {
     finalContent = `${finalContent.substring(0, CONTENT_LENGTH)} . . .`;
   }
   return (
     <Card.Body className="content-wrapper">
       <Heading level={4} className="h4">
-        <Highlight textObj={linkify(title)} highlight={highlightWords} />
+      {
+        linkifyText ? <Highlight textObj={linkify(title)} highlight={highlightWords} /> :
```
Comment (ddveloper) [inline client/src/components/Feed/Post.js]: Hi @aditilamkhede , could this be simplified as `<Highlight textObj={linkifyText ? linkify(title) : title} highlight={highlightWords} />` ?

File: client/src/components/Feed/Post.js [hunk 2]
```diff
@@ -842,18 +843,26 @@ const Post = ({
...
+        <Highlight textObj={title} highlight={highlightWords} />
+      }
+
       </Heading>
       <p className="post-description">
-        <Highlight textObj={linkify(finalContent)} highlight={highlightWords} />
+        {linkifyText ?
+          <Highlight textObj={linkify(finalContent)} highlight={highlightWords} />
```
Comment (ddveloper) [inline client/src/components/Feed/Post.js]: same here

Comment (aditilamkhede) [reply client/src/components/Feed/Post.js]: Yes, makes sense. will do that.

**Review by ddveloper — DISMISSED** [review]
LGTM! thanks @aditilamkhede

**Review by rabbi08 — APPROVED** [review]
Verified the fix as working fine, the link added in the post are displayed as link in the post details page. On the help board, it is displayed as text. Thanks.

Please refer attached video:

https://user-images.githubusercontent.com/67168400/104820700-ac5fb900-5804-11eb-871c-a45c928a6901.mov

**Comment by ddveloper** [comment]
Hi @aditilamkhede , I found 2 other places need set `linkify` to false, they are in `OrgPost.js` and `Activity.js`, maybe we can give a default `false` for the input `linkify` and only set it to be `true` in `PostPage.js`.

Besides, `linkify` itself sounds not straightforward, how about a name like `needLinkifyText`?

**Comment by aditilamkhede** [comment]
> Hi @aditilamkhede , I found 2 other places need set `linkify` to false, they are in `OrgPost.js` and `Activity.js`, maybe we can give a default `false` for the input `linkify` and only set it to be `true` in `PostPage.js`.
> 
> Besides, `linkify` itself sounds not straightforward, how about a name like `needLinkifyText`?

@ddvelope yes, Thanks will make these changes as well. I was also thinking there might be more places where POST component is being used. 

Name of the boolean variable is "linkifyText", I couldn't think of better name, "islinkifyText" wasn't a good choice either nor did "doLinkifyText". 

we can call it "linkifyText", "needLinkifyText", "isURL", "textToURL", "convertTextToURL"? let me know which one fits better with the functionality?

**Comment by ddveloper** [comment]
I think `convertTextToURL` is good 😃 , there won't be other `<Post>`, you can double confirm with a global search
image

**Comment by aditilamkhede** [comment]
> I think `convertTextToURL` is good 😃 , there won't be other `<Post>`, you can double confirm with a global search
> image
so the change the name of variable in POST component to "convertTextToURL", how about the function name linkify should we change that too?

**Comment by ddveloper** [comment]
function name `linkify` is good, sounds like a verb 👍

**Review by rabbi08 — DISMISSED** [review]
was looking for some verifications, only. : Done
Thanks.

**Comment by aditilamkhede** [comment]
@ddveloper @rabbi08 Please review with changes suggested by @ddveloper.

**Review by ddveloper — APPROVED** [review]
LGTM

**Review by rabbi08 — APPROVED** [review]
Thanks @aditilamkhede 
Verified the fix on feature branch, for Offer/Request post for Individual and Org post. The link text in the post is displayed as link on post description page only else it is displayed as text. 

Refer below GIF
ezgif com-video-to-gif-10

**Review by Zaydme — APPROVED** [review]
LGTM
Thank you @aditilamkhede

**Pull request merged**